doc/reports/spikes/ras-rm-account-removal.md

```diff
@@ -0,0 +1,55 @@
+
+
+  - Identify which respondent needs to be removed in `partysvc -> respondent`.  Make a note of the id, party_uuid and email address.
+  - In the `party_svc -> business_respondent` table, identify all of the `business_id`s that have a respondent_id that is the same as the id obtained in the previous step.
+
+   - Remove all of the records from `partysvc -> enrolment` where the respondent_id matches the id (from the first step).  There is a composite key of business_id and respondent_id between enrolment and business_respondent tables.
+   - Remove all of the records with the id from `partysvc -> business_respondent` -> respondent_id. The 'id' of the respondant is a foreign key of respondent id in the business_respondant table.
+   - Remove from `partysvc -> respondent` by email address.
+
+#### ras-rm-auth-service and django-oauth2-test
+Doing the above will make it so that the user cannot log in.  To remove the record of the user from the system, you also have to do the following
+ - Remove from `auth -> user` by email address (if using ras-rm-auth-service)
+- Remove from `public -> credentials_oauthuser` by email address (if using django-oauth service)
+
+#### ras-collection-exercise
+This service shouldn't have any issues if an account gets deleted. It goes to the party service for a business id.  If this has been deleted because of the previous step then it will fail.  Failure (from what I can tell) means that ultimately a SEFT survey (or collection instrument) can't be uploaded.  This is probably fine as a business would need a respondent with an account in order to be able to upload one, if there was only 1 and we deleted it, then they can't get to that screen.  If they have multiple respondents with accounts (is this possible?) then there will be a record in the business table for the other respondent so it should succeed.
+
+#### rm-sample-service
+Posts to the /parties endpoint.  This endpoint returns a business so this shouldn't be an issue as I think this all happens before the users account gets created at the start of the process.
+
+#### rm-action-service
+This service goes to the party service to get information.  If there is never an action involving an account that's been deleted then it should be fine.  Need to find out a few things:
+  - What happens if it can't find the party?
+  - What provides the action service with stuff to do?  If we can find that out then we can see if it would ever do an action with a party that doesn't exist any more. 
+
+#### rm-reporting
+From the looks of things, this should be unaffected.  If the party_id isn't found in the search, then it'll just come back with nothing.  Whether or not that's the desired behaviour is another question though.  This might link in with what in in the below section about whether or not we need to make sure the respondent has completed all the surveys they need to.
```

Comment:
> Whether or not that's the desired behaviour is another question though.

This should be fine.  If all enrolments are removed from the respondent then the search will treat as a business that has nobody enrolled to complete the surveys (like a lot of newly selected companies are).